testing/supporting_docs/ui_selectors_and_their_importance.md

```diff
@@ -0,0 +1,178 @@
+can be more prominent for hybrid apps or in browser testing where xpaths can become
+ much longer. 
+
+It is possible to improve the rigidness and performance of xpaths in test by using 
+relative xpaths that are shorter and more flexible.  For example, a standard xpath 
+selector from an Android application such as:
+
+    "//android.widget.TextView[@text='My Messages']"
+
+Can be shortened to:
+
+	"//TextView[@text='My Messages']"
+
+The only problem with a relative xpath compared to an accessibilityId, id, or name 
+is that when using one like the example above there may also a dependency on the 
+text value of the page.  If the text on the page is changed, then it will be required 
+to change the selector in the test as well.   While this is an improvement still, one 
+risk with using relative xpaths is over time, as developers change, the knowledge of 
+the importance of relative xpaths in providing benefits to the test suite can be lost, 
+resulting inefficient and more brittle xpaths, as produced by tools such as “Appium 
+Inspector”, being used and effecting test suite performance and reliability. 
+
+## Examples of adding Adding Selectors to Applications
+
+### iOS (AccessibilityId Example)
+
+By default, if an ‘accessibilityId’ is not defined for an element, an iOS application will 
+dynamically set the ‘accessibilitityId’ of an element to the value of the element. This is not 
+ideal when testing for any elements that display information that will be dynamically created, 
+such as a username for a user, or possibly with elements such as toggle switches or checkboxes 
+where the value is updated after being clicked.
+
+For iOS the ‘accessibilityId’ can be added to a UI component in two ways. When viewing the 
+storyboards in Xcode, to add an ‘accessibilityId’, firstly select the element, then open the 
+identity inspector tab. Under the ‘User Defined Runtime Attributes’, add a new Key Path called 
+‘accessibilityIdentifier’, set the Type to ‘String’, and set the Value to the desired name of 
+the new selector.
+
+
+![img text](./images/adding_accessibilityid_xcode_identity_inspector_panel.png "Adding AccessibilityId Via Xcode Identity Inspector Panel")
+
+The second way of adding an ‘accessibilityId’ to a UI element is to do so through Swift code.  
+This can be done by setting the ‘accessibilityIdentifier’ attribute of a UI element as shown 
+below.
+
+```swift
+if (indexPath.row == 0) {
+    fieldNameLabel.text = "Name"
+    fieldValueLabel.text = self.currentUser!.fullName
+    fieldValueLabel.accessibilityIdentifier = "username"
+} else {
+    fieldName.text = "Email"
+    fieldValueLabel.text = self.currentUser!.email
+    fieldValueLabel.accessibilityIdentifier = "email"
+}
+```
+
+### Android (AccessibilityId Example)
+Unlike iOS Android does not dynamically assign any values to the AccesibilityId.  Instead if 
+the developer wishes to use AccessibilityId for testing purposes, it can be added to Android 
```

Comment:
@twnolan consistency should be persisted with the word 'android' across the whole document, e.g. starting with either lower or upper case.